contracts/PeeranhaToken.sol

```diff
@@ -4,45 +4,84 @@ import "./libraries/RewardLib.sol";
-import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20CappedUpgradeable.sol";
 
-contract PeeranhaToken is ERC20Upgradeable, ERC20PausableUpgradeable, ERC20CappedUpgradeable {
+contract PeeranhaToken is IPeeranhaToken, ERC20Upgradeable, ERC20PausableUpgradeable, OwnableUpgradeable {
   uint256 public constant FRACTION = (10 ** 18);
-  uint256 public constant TOTAL_SUPPLY = 1000000000 * FRACTION;
-  uint256 public constant REWARD_WEEK = 1000000;
+  // uint256 public constant TOTAL_SUPPLY = 1000000000 * FRACTION;
+  uint256 public constant REWARD_WEEK = 100000;
+  uint256 public constant USER_REWARD = 100;
+
+  
+  ///
+  // 100 000  - 100 token * 1000 user
+  // 40 000 000 - company
+  ///
+  // TODO: add actions that allow owner to mint 40% of total token supply
+  // rewrite transfer - boost, balanceOf - boost
+  // add method getUserStakedBalance
+  // getstakedbalance (transfer + get balance)
+  ///
+
   uint256 public constant ACTIVE_USERS_IN_PERIOD = 1000;
 
   mapping(uint16 => uint256) poolTokens;
 
   TokenLib.StatusRewardContainer statusRewardContainer;
+  TokenLib.UserPeriodStake userPeriodStake;
+  TokenLib.StakeTotalContainer stakeTotalContainer;
   IPeeranha peeranha;
 
   event GetReward(address user, uint16 period);
+  event SetStake(address user, uint16 period, uint256 stake);
+
 
   function initialize(string memory name, string memory symbol, address peeranhaNFTContractAddress) public initializer {
-    __Token_init(name, symbol, TOTAL_SUPPLY);
+    __Token_init(name, symbol);
     peeranha = IPeeranha(peeranhaNFTContractAddress);
+    __Ownable_init_unchained();
   }
 
-  function __Token_init(string memory name, string memory symbol, uint256 cap) internal initializer {
+  function __Token_init(string memory name, string memory symbol) internal initializer {
     __ERC20_init_unchained(name, symbol);
     __Pausable_init_unchained();
-    __ERC20Capped_init_unchained(cap);
     __ERC20Pausable_init_unchained();
     __Token_init_unchained();
   }
 
   function __Token_init_unchained() internal initializer {
   }
 
-  function _beforeTokenTransfer(address from, address to, uint256 amount) internal virtual override(ERC20Upgradeable, ERC20PausableUpgradeable, ERC20CappedUpgradeable) {
+  function _beforeTokenTransfer(address from, address to, uint256 amount) internal virtual override(ERC20Upgradeable, ERC20PausableUpgradeable/*, ERC20CappedUpgradeable*/) {
     super._beforeTokenTransfer(from, to, amount);
   }
 
+  function stakeBalanceOf(address account) external view returns(uint256) { // unitTest
+    TokenLib.StakeUserContainer storage stakeUserContainer = userPeriodStake.userPeriodStake[account];
+    uint16 period = RewardLib.getPeriod(CommonLib.getTimestamp()) + 1;
+
+    uint256 stakedToken;
+    (uint256 correctPeriod, bool status) = findInternal(stakeUserContainer.stakeChangePeriods, 0, stakeUserContainer.stakeChangePeriods.length, period);
```

Comment:
Why this `findInternal` is needed? Can't we just check stake at the last period?